mindee/product/generated/generated_v1_prediction.py

```diff
@@ -0,0 +1,77 @@
+import re
+from typing import Dict, List, Union
+
+from mindee.parsing.common import Prediction, StringDict, clean_out_string
+from mindee.parsing.custom import GeneratedObjectField
+from mindee.parsing.custom.generated_list import GeneratedListField
+from mindee.parsing.standard.text import StringField
+
+
+class GeneratedV1Prediction(Prediction):
+    """Generated V1 document prediction results."""
+
+    fields: Dict[str, Union[GeneratedListField, StringField, GeneratedObjectField]]
+    """Dictionary of all fields in the document"""
+
+    def __init__(self, raw_prediction: StringDict) -> None:
+        """
+        Generated document.
+
+        :param raw_prediction: Dictionary containing the JSON document response
+        """
+        super().__init__(raw_prediction)
+        self.fields = {}
+
+    def __str__(self) -> str:
+        out_str = ""
+        pattern = re.compile(r"^(\n*[  ]*)( {2}):")
+        for field_name, field_value in self.fields.items():
+            str_value = ""
+            if (
+                isinstance(field_value, GeneratedListField)
+                and len(field_value.values) > 0
+            ):
+                if isinstance(field_value.values[0], GeneratedObjectField):
+                    str_value += re.sub(
+                        pattern, r"\1* :", f"{field_value.values[0].__str__(1)}"
+                    )
+                else:
+                    str_value += re.sub(pattern, r"\1* :", f"{field_value.values[0]}\n")
+                for sub_value in field_value.values[1:]:
+                    if isinstance(sub_value, GeneratedObjectField):
+                        str_value += re.sub(pattern, r"\1* :", sub_value.__str__(1))
```

Comment:
Don't call `__str__` directly, and don't change the method signature of a magic method.